### PR TITLE
Maverick registration

### PR DESCRIPTION
**Pull request by hhsecond**
## What does this PR do?

Register maverick from CLI

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda

### CODE REVIEW

File: src/lightning/app/utilities/clusters.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def _ensure_cluster_project_binding(client: LightningClient, project_id: str, cl
 
     client.projects_service_create_project_cluster_binding(
         project_id=project_id,
-        body=V1ProjectClusterBinding(cluster_id=cluster_id, project_id=project_id),
+        body=ProjectIdProjectclustersbindingsBody(cluster_id=cluster_id),
```
Comment (hhsecond) [inline src/lightning/app/utilities/clusters.py:20]: not sure if this was ever worked. The input types were wrong here

File: src/lightning/app/cli/connect/maverick.py [hunk 2]
```diff
@@ -194,6 +211,55 @@ def connect_maverick(name: str) -> None:
...
+            raise ValueError(f"Project {project_name} does not exist.")
+    else:
+        project_id = _get_project(client, verbose=False).project_id
+
+    cluster_bindings = client.projects_service_list_project_cluster_bindings(project_id=project_id)
+    for c in cluster_bindings.clusters:
+        if c.cluster_name == name:
+            existing_cluster = client.cluster_service_get_cluster(id=c.cluster_id)
+            if existing_cluster.status.phase == V1ClusterState.RUNNING:
+                raise RuntimeError(f"Cluster {name} already exists and is running.")
```
Comment (tchaton) [inline src/lightning/app/cli/connect/maverick.py:242]: Let's not raise RuntimeError in a CLI. Let's use the error_and_exit helper.

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/lightning/app/cli/connect/maverick.py (hunk 2 above)
Comment (hhsecond) [reply src/lightning/app/cli/connect/maverick.py:242]: This is being caught by the caller and use `rich.print` to pass the message to the user

File: src/lightning/app/cli/connect/maverick.py [hunk 3]
```diff
@@ -66,7 +77,8 @@ def get_lightning_daemon_command(prefix: str) -> str:
 
 
 @click.argument("name", required=True)
-def connect_maverick(name: str) -> None:
+@click.option("--project_name", help="The project name to which the machine should connect..", required=False)
```
Comment (ethanwharris) [inline src/lightning/app/cli/connect/maverick.py:80]:
```suggestion
@click.option("--project_name", help="The project name to which the machine should connect.", required=False)
```

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Pull request merged**